src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java

```diff
@@ -14,30 +15,21 @@ public TransactionalMemoizationStore(MemoizationStore memoizationStore) {
 
     @Override
     public void put(Integer start, Integer size, byte[] value) {
-        if (!stagedHashes.containsKey(start)) {
-            stagedHashes.put(start, new HashMap<>());
-        }
-
-        HashMap<Integer, byte[]> hashesBySize = stagedHashes.get(start);
-        hashesBySize.put(size, value);
+        stagedHashes.put(Pair.of(start, size), value);
     }
 
     @Override
     public byte[] get(Integer start, Integer size) {
-        byte[] hash = null;
-
-        if (stagedHashes.containsKey(start)) {
-            hash = stagedHashes.get(start).get(size);
+        if (stagedHashes.containsKey(Pair.of(start, size)))  {
```

Comment:
if we assume that there is not going to be a `null` hash value for any star, size pair we can even compress it to a 1 liner, something like this:
```
public byte[] get(Integer start, Integer size) {
        return stagedHashes.getOrDefault(Pair.of(start, size), memoizationStore.get(start, size));
    }
```